core/metric.go

```diff
@@ -45,8 +53,102 @@ type Metric interface {
 	Unit() string
 }
 
+// NamespaceElement defines the method set for the namespace elements
+type NamespaceElement interface {
+	IsDynamic() bool
+	Value() string
+	Name() string
+	Description() string
+	SetValue(value string)
+	SetName(name string)
+	SetDescription(desc string)
+}
+
+// Namespace defines array of namespace elements
 type Namespace []NamespaceElement
 
+// DynamicElement defines the dynamic element of a namespace
+type DynamicElement struct {
+	Value_       string
+	Description_ string
+	Name_        string
+}
```

Comment:
Agreed as well but they were unexported but what I learned is that GOB needs all fields to be exported. As I thought there is GOB issue, I exposed all of them now. Correct me if I miss anything.